pysteps/xarray_helpers.py

```diff
@@ -142,25 +169,34 @@ def convert_input_to_xarray_dataset(
 
     data_vars = {
         var_name: (
-            ["y", "x"],
+            dims,
             precip,
             {
                 "units": attrs["units"],
                 "standard_name": attrs["standard_name"],
                 "long_name": attrs["long_name"],
                 "grid_mapping": "projection",
-                "transform": metadata["transform"],
-                "accutime": metadata["accutime"],
-                "threshold": metadata["threshold"],
-                "zerovalue": metadata["zerovalue"],
-                "zr_a": metadata["zr_a"],
-                "zr_b": metadata["zr_b"],
             },
         )
     }
+
+    metadata_keys = [
+        "transform",
+        "accutime",
+        "threshold",
+        "zerovalue",
+        "zr_a",
+        "zr_b",
+        "cartesian_unit",
```

Comment:
Oh actually one more thing: the cartesian unit is also already present in the attributes of the x and y dimension

Reply:
Done